src/chanjo2/crud/intervals.py

```diff
@@ -42,12 +42,21 @@ def _filter_intervals_by_ensembl_ids(
     return intervals.filter(interval_type.ensembl_id.in_(ensembl_ids))
 
 
+def _filter_intervals_by_ensembl_gene_ids(
+    intervals: query.Query,
+    interval_type: Union[SQLGene, SQLTranscript, SQLExon],
+    ensembl_gene_ids: List[str],
+) -> List[Union[SQLGene, SQLTranscript, SQLExon]]:
```

Comment:
```suggestion
) -> query.Query:
```
I do not think you are reading the db yet and interpolating to real objects